runtime/development/src/lib.rs

```diff
@@ -936,6 +936,56 @@ impl PoolUpdateGuard for UpdateGuard {
 	}
 }
 
+pub struct CurrencyPriceSource;
+impl CurrencyPrice<CurrencyId> for CurrencyPriceSource {
+	type Rate = Rate;
+	type Moment = Moment;
+
+	fn get_latest(
+		base: CurrencyId,
+		quote: Option<CurrencyId>,
+	) -> Option<PriceValue<CurrencyId, Self::Rate, Self::Moment>> {
+		match base {
+			CurrencyId::Tranche(pool_id, tranche_id) => {
+				let now = <pallet_timestamp::Pallet<Runtime> as UnixTime>::now().as_secs();
+				let mut pool = pallet_pools::Pool::<Runtime>::get(pool_id)?;
```

Comment:
Should the body of this be moved into the pool pallet (a `get_tranche_token_price` method or something like that?)

Reply:
I think that makes sense, yeah 👍 Will move it.